[TeamMDPolicy] Passing reference of variable on stack?

With clang-tidy, the following warning is caused:
```
warning: -lkokkoscore: 'linker' input unused [clang-diagnostic-unused-command-line-argument]
/usr/local/include/impl/Kokkos_TeamMDPolicy.hpp:107:3: warning: Address of stack memory associated with local variable 'team' is still referred to by a temporary object on the stack upon returning to the caller.  This will be a dangling reference [clang-analyzer-core.StackAddressEscape]
  107 |   return TeamThreadRange(team, count);
      |   ^
/home/vscode/clang_tidy_teammdrange_repro.cpp:11:9: note: Calling 'parallel_for<Kokkos::TeamPolicy<Kokkos::Serial>, (lambda at /home/vscode/clang_tidy_teammdrange_repro.cpp:14:13)>'
   11 |         Kokkos::parallel_for(
      |         ^~~~~~~~~~~~~~~~~~~~~
   12 |             "teammdrange_repro",
      |             ~~~~~~~~~~~~~~~~~~~~
   13 |             Kokkos::TeamPolicy<Exec>(1, Kokkos::AUTO),
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   14 |             KOKKOS_LAMBDA(const Kokkos::TeamPolicy<Exec>::member_type& team) {
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   15 |                 // Use TeamThreadMDRange so nested_policy() returns TeamThreadRange
      |                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   16 |                 // (clang-analyzer-core.StackAddressEscape warning)
      |                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   17 |                 Kokkos::parallel_for(Kokkos::TeamThreadMDRange(team, dim0, dim1),
      |                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   18 |                                      [&](std::uint64_t i, std::uint64_t j) {
      |                                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   19 |                                          (void)i;
      |                                          ~~~~~~~~
   20 |                                          (void)j;
      |                                          ~~~~~~~~
   21 |                                      });
      |                                      ~~~
   22 |             });
      |             ~~
/usr/local/include/Kokkos_Parallel.hpp:135:3: note: Calling 'ParallelFor::execute'
  135 |   closure.execute();
      |   ^~~~~~~~~~~~~~~~~
/usr/local/include/Serial/Kokkos_Serial_Parallel_Team.hpp:263:5: note: Calling 'ParallelFor::exec'
  263 |     this->template exec<typename Policy::work_tag>(
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  264 |         internal_instance->m_thread_team_data);
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/Serial/Kokkos_Serial_Parallel_Team.hpp:226:27: note: Assuming 'ileague' is < field 'm_league'
  226 |     for (int ileague = 0; ileague < m_league; ++ileague) {
      |                           ^~~~~~~~~~~~~~~~~~
/usr/local/include/Serial/Kokkos_Serial_Parallel_Team.hpp:226:5: note: Loop condition is true.  Entering loop body
  226 |     for (int ileague = 0; ileague < m_league; ++ileague) {
      |     ^
/usr/local/include/Serial/Kokkos_Serial_Parallel_Team.hpp:227:7: note: Calling 'operator()'
  227 |       m_functor(Member(data, ileague, m_league));
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/vscode/clang_tidy_teammdrange_repro.cpp:17:17: note: Calling 'parallel_for<Kokkos::Rank<2, Kokkos::Iterate::Default>, Kokkos::Impl::HostThreadTeamMember<Kokkos::Serial>, (lambda at /home/vscode/clang_tidy_teammdrange_repro.cpp:18:38)>'
   17 |                 Kokkos::parallel_for(Kokkos::TeamThreadMDRange(team, dim0, dim1),
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   18 |                                      [&](std::uint64_t i, std::uint64_t j) {
      |                                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   19 |                                          (void)i;
      |                                          ~~~~~~~~
   20 |                                          (void)j;
      |                                          ~~~~~~~~
   21 |                                      });
      |                                      ~~
/usr/local/include/Kokkos_ExecPolicy.hpp:1161:3: note: Calling 'md_parallel_impl<Kokkos::Rank<2, Kokkos::Iterate::Default>, Kokkos::TeamThreadMDRange<Kokkos::Rank<2, Kokkos::Iterate::Default>, Kokkos::Impl::HostThreadTeamMember<Kokkos::Serial>>, (lambda at /home/vscode/clang_tidy_teammdrange_repro.cpp:18:38), Kokkos::Impl::NoReductionTag>'
 1161 |   Impl::md_parallel_impl<Rank>(policy, lambda, Impl::NoReductionTag());
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/impl/Kokkos_TeamMDPolicy.hpp:234:9: note: 'par_thread' is equal to ParThread
  234 |       ((TeamMDPolicy::par_thread == Impl::TeamMDRangeParThread::ParThread) &&
      |         ^~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/impl/Kokkos_TeamMDPolicy.hpp:234:8: note: Left side of '&&' is true
  234 |       ((TeamMDPolicy::par_thread == Impl::TeamMDRangeParThread::ParThread) &&
      |        ^
/usr/local/include/impl/Kokkos_TeamMDPolicy.hpp:235:9: note: 'par_vector' is not equal to ParVector
  235 |        (TeamMDPolicy::par_vector == Impl::TeamMDRangeParVector::ParVector))
      |         ^~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/impl/Kokkos_TeamMDPolicy.hpp:234:7: note: '?' condition is false
  234 |       ((TeamMDPolicy::par_thread == Impl::TeamMDRangeParThread::ParThread) &&
      |       ^
/usr/local/include/impl/Kokkos_TeamMDPolicy.hpp:239:8: note: 'iter' is not equal to Right
  239 |       (iter == Iterate::Right) ? 0 : (total_nest_level - 1);
      |        ^~~~
/usr/local/include/impl/Kokkos_TeamMDPolicy.hpp:239:7: note: '?' condition is false
  239 |       (iter == Iterate::Right) ? 0 : (total_nest_level - 1);
      |       ^
/usr/local/include/impl/Kokkos_TeamMDPolicy.hpp:246:8: note: 'par_thread' is equal to ParThread
  246 |       (TeamMDPolicy::par_thread == TeamMDRangeParThread::ParThread)
      |        ^~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/impl/Kokkos_TeamMDPolicy.hpp:246:7: note: '?' condition is true
  246 |       (TeamMDPolicy::par_thread == TeamMDRangeParThread::ParThread)
      |       ^
/usr/local/include/impl/Kokkos_TeamMDPolicy.hpp:250:8: note: 'par_vector' is not equal to ParVector
  250 |       (TeamMDPolicy::par_vector == TeamMDRangeParVector::ParVector)
      |        ^~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/impl/Kokkos_TeamMDPolicy.hpp:250:7: note: '?' condition is false
  250 |       (TeamMDPolicy::par_vector == TeamMDRangeParVector::ParVector)
      |       ^
/usr/local/include/impl/Kokkos_TeamMDPolicy.hpp:260:3: note: Calling 'nested_loop<Kokkos::Rank<2, Kokkos::Iterate::Default>, 0, -2, 1, Kokkos::TeamThreadMDRange<Kokkos::Rank<2, Kokkos::Iterate::Default>, Kokkos::Impl::HostThreadTeamMember<Kokkos::Serial>>, (lambda at /home/vscode/clang_tidy_teammdrange_repro.cpp:18:38), Kokkos::Impl::NoReductionTag, >'
  260 |   nested_loop(InitTeamMDMode(), InitNestingTracker(), policy, lambda,
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  261 |               std::forward<ReductionValueType>(val));
      |               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/impl/Kokkos_TeamMDPolicy.hpp:168:27: note: 'outer_direction' is not equal to Right
  168 |       CurrentNestLevel + (Rank::outer_direction == Iterate::Right ? 1 : -1);
      |                           ^~~~~~~~~~~~~~~~~~~~~
/usr/local/include/impl/Kokkos_TeamMDPolicy.hpp:168:27: note: '?' condition is false
/usr/local/include/impl/Kokkos_TeamMDPolicy.hpp:174:3: note: Loop condition is true.  Entering loop body
  174 |   for (int i = 0; i != policy.boundaries[CurrentNestLevel]; ++i) {
      |   ^
/usr/local/include/impl/Kokkos_TeamMDPolicy.hpp:177:19: note: 'outer_direction' is not equal to Right
  177 |     if constexpr (Rank::outer_direction == Iterate::Right) {
      |                   ^~~~~~~~~~~~~~~~~~~~~
/usr/local/include/impl/Kokkos_TeamMDPolicy.hpp:177:5: note: Taking false branch
  177 |     if constexpr (Rank::outer_direction == Iterate::Right) {
      |     ^
/usr/local/include/impl/Kokkos_TeamMDPolicy.hpp:181:7: note: Calling 'nested_loop<Kokkos::Impl::TeamMDRangeParThread::ParThread, Kokkos::Impl::TeamMDRangeParVector::NotParVector, Kokkos::Rank<2, Kokkos::Iterate::Default>, 0, -2, 0, Kokkos::TeamThreadMDRange<Kokkos::Rank<2, Kokkos::Iterate::Default>, Kokkos::Impl::HostThreadTeamMember<Kokkos::Serial>>, (lambda at /home/vscode/clang_tidy_teammdrange_repro.cpp:18:38), Kokkos::Impl::NoReductionTag, int>'
  181 |       nested_loop(TeamMDNextMode(), NextNestingTracker(), policy, lambda,
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  182 |                   std::forward<ReducerValueType>(val), i, args...);
      |                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/impl/Kokkos_TeamMDPolicy.hpp:200:27: note: 'outer_direction' is not equal to Right
  200 |       CurrentNestLevel + (Rank::outer_direction == Iterate::Right ? 1 : -1);
      |                           ^~~~~~~~~~~~~~~~~~~~~
/usr/local/include/impl/Kokkos_TeamMDPolicy.hpp:200:27: note: '?' condition is false
/usr/local/include/impl/Kokkos_TeamMDPolicy.hpp:209:7: note: Calling 'nested_policy<Kokkos::Impl::HostThreadTeamMember<Kokkos::Serial>>'
  209 |       nested_policy(mode, policy.team, policy.boundaries[CurrentNestLevel]),
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/impl/Kokkos_TeamMDPolicy.hpp:107:3: note: Address of stack memory associated with local variable 'team' is still referred to by a temporary object on the stack upon returning to the caller.  This will be a dangling reference
  106 |     TeamHandle const team, int count) {
      |     ~~~~~~~~~~~~~~~~~~~~~
  107 |   return TeamThreadRange(team, count);
      |   ^
```

`team` of `nested_loop` is passed by value, but `thread` of `TeamThreadRangeBoundariesStruct`, which is passed to, has reference type.
Is it OK to ignore?

1. Compiler:
  - Used for Building Kokkos: gcc (Ubuntu 13.1.0-8ubuntu1~22.04) 13.1.0
  - Used for clang-tidy check: Ubuntu LLVM version 20.1.8  Optimized build.
2. Kokkos release: Upstream develop branch ( https://github.com/kokkos/kokkos/commit/1707507d7682bf77f0ed02a2fd51d6f707514c09 )
3. Platform: Ubuntu 22.04 x86_64
4. CMake Configure Command: `cmake ..`
5. Minimum code for reproducing warniing
```
#include <Kokkos_Core.hpp>
#include <cstdint>

int main(int argc, char** argv) {
    Kokkos::initialize(argc, argv);
    {
        using Exec = Kokkos::DefaultHostExecutionSpace;
        constexpr std::uint64_t dim0 = 4;
        constexpr std::uint64_t dim1 = 8;

        Kokkos::parallel_for(
            "teammdrange_repro",
            Kokkos::TeamPolicy<Exec>(1, Kokkos::AUTO),
            KOKKOS_LAMBDA(const Kokkos::TeamPolicy<Exec>::member_type& team) {
                // Use TeamThreadMDRange so nested_policy() returns TeamThreadRange
                // (clang-analyzer-core.StackAddressEscape warning)
                Kokkos::parallel_for(Kokkos::TeamThreadMDRange(team, dim0, dim1),
                                     [&](std::uint64_t i, std::uint64_t j) {
                                         (void)i;
                                         (void)j;
                                     });
            });
    }
    Kokkos::finalize();
    return 0;
}
```
6. Command line needed to reproduce the warning: `clang-tidy-20 --extra-arg=-std=c++23 --extra-arg=-lkokkoscore clang_tidy_teammdrange_repro.cpp`
7.  `KokkosCore_config.h` header file (generated during the build)
```
#if !defined(KOKKOS_MACROS_HPP) || defined(KOKKOS_CORE_CONFIG_H)
#error \
    "Do not include KokkosCore_config.h directly; include Kokkos_Macros.hpp instead."
#else
#define KOKKOS_CORE_CONFIG_H
#endif

// KOKKOS_VERSION % 100 is the patch level
// KOKKOS_VERSION / 100 % 100 is the minor version
// KOKKOS_VERSION / 10000 is the major version
#define KOKKOS_VERSION 50099
/* #undef below */
// yields /* undef KOKKOS_VERSION_FOO */
#define KOKKOS_VERSION_MAJOR 5
#define KOKKOS_VERSION_MINOR 0
#define KOKKOS_VERSION_PATCH 99

/* #undef KOKKOS_IMPL_BUILD_SHARED_LIBS */

/* Execution Spaces */
#define KOKKOS_ENABLE_SERIAL
/* #undef KOKKOS_ENABLE_OPENMP */
/* #undef KOKKOS_ENABLE_OPENACC */
/* #undef KOKKOS_ENABLE_THREADS */
/* #undef KOKKOS_ENABLE_CUDA */
/* #undef KOKKOS_ENABLE_HIP */
/* #undef KOKKOS_ENABLE_HPX */
/* #undef KOKKOS_ENABLE_SYCL */

/* General Settings */
#define KOKKOS_ENABLE_CXX20
/* #undef KOKKOS_ENABLE_CXX23 */
/* #undef KOKKOS_ENABLE_CXX26 */

/* #undef KOKKOS_ENABLE_CUDA_RELOCATABLE_DEVICE_CODE */
/* #undef KOKKOS_ENABLE_CUDA_UVM */
/* #undef KOKKOS_ENABLE_CUDA_LAMBDA */
/* #undef KOKKOS_ENABLE_CUDA_CONSTEXPR */
/* #undef KOKKOS_ENABLE_IMPL_CUDA_MALLOC_ASYNC */
/* #undef KOKKOS_ENABLE_IMPL_CUDA_UNIFIED_MEMORY */
/* #undef KOKKOS_ENABLE_HIP_RELOCATABLE_DEVICE_CODE */
/* #undef KOKKOS_ENABLE_HIP_MULTIPLE_KERNEL_INSTANTIATIONS */
/* #undef KOKKOS_ENABLE_IMPL_HIP_MALLOC_ASYNC */
/* #undef KOKKOS_ENABLE_SYCL_RELOCATABLE_DEVICE_CODE */
/* #undef KOKKOS_ENABLE_IMPL_SYCL_OUT_OF_ORDER_QUEUES */
/* #undef KOKKOS_IMPL_SYCL_DEVICE_GLOBAL_SUPPORTED */
/* #undef KOKKOS_ENABLE_OPENACC_FORCE_HOST_AS_DEVICE */
/* #undef KOKKOS_ENABLE_IMPL_HPX_ASYNC_DISPATCH */
/* #undef KOKKOS_ENABLE_DEBUG */
#define KOKKOS_ENABLE_DEBUG_DUALVIEW_MODIFY_CHECK  // deprecated
/* #undef KOKKOS_ENABLE_DEBUG_BOUNDS_CHECK */
/* #undef KOKKOS_ENABLE_TUNING */
/* #undef KOKKOS_ENABLE_DEPRECATED_CODE_4 */
#define KOKKOS_ENABLE_DEPRECATED_CODE_5
#define KOKKOS_ENABLE_DEPRECATION_WARNINGS
/* #undef KOKKOS_ENABLE_LARGE_MEM_TESTS */
#define KOKKOS_ENABLE_COMPLEX_ALIGN
/* #undef KOKKOS_OPT_RANGE_AGGRESSIVE_VECTORIZATION */
/* #undef KOKKOS_ENABLE_AGGRESSIVE_VECTORIZATION */
/* #undef KOKKOS_ENABLE_EXPERIMENTAL_CXX20_MODULES */
#define KOKKOS_ENABLE_IMPL_MDSPAN
/* #undef KOKKOS_ENABLE_IMPL_VIEW_LEGACY */
#define KOKKOS_ENABLE_IMPL_REF_COUNT_BRANCH_UNLIKELY
/* #undef KOKKOS_ENABLE_IMPL_VIEW_OF_VIEWS_DESTRUCTOR_PRECONDITION_VIOLATION_WORKAROUND */
/* #undef KOKKOS_ENABLE_ATOMICS_BYPASS */
/* #undef KOKKOS_ENABLE_IMPL_CHECK_POSSIBLY_BREAKING_LAYOUTS */
/* #undef KOKKOS_ENABLE_BENCHMARKS_HEAVY */

/* TPL Settings */
/* #undef KOKKOS_ENABLE_HWLOC */
#define KOKKOS_ENABLE_LIBDL
/* #undef KOKKOS_ENABLE_LIBQUADMATH */
/* #undef KOKKOS_ENABLE_ONEDPL */
/* #undef KOKKOS_ENABLE_ROCTHRUST */

/* Embedded dependencies */
#define KOKKOS_IMPL_DESUL_VERSION "5e8dd8b549c0de391acb77535a40aa5668c35d01"
#define KOKKOS_IMPL_MDSPAN_VERSION "5d4eb209c77f4744980c0b0c2af44636cc81b08b"

/* #undef KOKKOS_ARCH_ARMV80 */
/* #undef KOKKOS_ARCH_ARMV8_THUNDERX */
/* #undef KOKKOS_ARCH_ARMV81 */
/* #undef KOKKOS_ARCH_ARMV8_THUNDERX2 */
/* #undef KOKKOS_ARCH_ARMV9_GRACE */
/* #undef KOKKOS_ARCH_ARMV84 */
/* #undef KOKKOS_ARCH_ARMV84_SVE */
/* #undef KOKKOS_ARCH_ARM_NEON */
/* #undef KOKKOS_ARCH_ARM_SVE */
/* #undef KOKKOS_ARCH_A64FX */
/* #undef KOKKOS_ARCH_AVX */
/* #undef KOKKOS_ARCH_AVX2 */
/* #undef KOKKOS_ARCH_AVX512XEON */
/* #undef KOKKOS_ARCH_KNC */
/* #undef KOKKOS_ARCH_AVX512MIC */
/* #undef KOKKOS_ARCH_POWER7 */
/* #undef KOKKOS_ARCH_POWER8 */
/* #undef KOKKOS_ARCH_POWER9 */
/* #undef KOKKOS_ARCH_RISCV_SG2042 */
/* #undef KOKKOS_ARCH_RISCV_RVA22V */
/* #undef KOKKOS_ARCH_RISCV_U74MC */
/* #undef KOKKOS_ARCH_INTEL_GEN */
/* #undef KOKKOS_ARCH_INTEL_DG1 */
/* #undef KOKKOS_ARCH_INTEL_DG2 */
/* #undef KOKKOS_ARCH_INTEL_GEN9 */
/* #undef KOKKOS_ARCH_INTEL_GEN11 */
/* #undef KOKKOS_ARCH_INTEL_GEN12LP */
/* #undef KOKKOS_ARCH_INTEL_XEHP */
/* #undef KOKKOS_ARCH_INTEL_PVC */
/* #undef KOKKOS_ARCH_INTEL_GPU */
/* #undef KOKKOS_ARCH_KEPLER */
/* #undef KOKKOS_ARCH_KEPLER30 */
/* #undef KOKKOS_ARCH_KEPLER32 */
/* #undef KOKKOS_ARCH_KEPLER35 */
/* #undef KOKKOS_ARCH_KEPLER37 */
/* #undef KOKKOS_ARCH_MAXWELL */
/* #undef KOKKOS_ARCH_MAXWELL50 */
/* #undef KOKKOS_ARCH_MAXWELL52 */
/* #undef KOKKOS_ARCH_MAXWELL53 */
/* #undef KOKKOS_ARCH_PASCAL */
/* #undef KOKKOS_ARCH_PASCAL60 */
/* #undef KOKKOS_ARCH_PASCAL61 */
/* #undef KOKKOS_ARCH_VOLTA */
/* #undef KOKKOS_ARCH_VOLTA70 */
/* #undef KOKKOS_ARCH_VOLTA72 */
/* #undef KOKKOS_ARCH_TURING75 */
/* #undef KOKKOS_ARCH_AMPERE */
/* #undef KOKKOS_ARCH_AMPERE80 */
/* #undef KOKKOS_ARCH_AMPERE86 */
/* #undef KOKKOS_ARCH_AMPERE87 */
/* #undef KOKKOS_ARCH_ADA89 */
/* #undef KOKKOS_ARCH_HOPPER */
/* #undef KOKKOS_ARCH_HOPPER90 */
/* #undef KOKKOS_ARCH_BLACKWELL */
/* #undef KOKKOS_ARCH_BLACKWELL100 */
/* #undef KOKKOS_ARCH_BLACKWELL103 */
/* #undef KOKKOS_ARCH_BLACKWELL120 */
/* #undef KOKKOS_ARCH_AMD_ZEN */
/* #undef KOKKOS_ARCH_AMD_ZEN2 */
/* #undef KOKKOS_ARCH_AMD_ZEN3 */
/* #undef KOKKOS_ARCH_AMD_ZEN4 */
/* #undef KOKKOS_ARCH_AMD_ZEN5 */
/* #undef KOKKOS_ARCH_AMD_GFX906 */
/* #undef KOKKOS_ARCH_AMD_GFX908 */
/* #undef KOKKOS_ARCH_AMD_GFX90A */
/* #undef KOKKOS_ARCH_AMD_GFX940 */
/* #undef KOKKOS_ARCH_AMD_GFX942 */
/* #undef KOKKOS_ARCH_AMD_GFX942_APU */
/* #undef KOKKOS_ARCH_AMD_GFX950 */
/* #undef KOKKOS_ARCH_AMD_GFX1030 */
/* #undef KOKKOS_ARCH_AMD_GFX1100 */
/* #undef KOKKOS_ARCH_AMD_GFX1103 */
/* #undef KOKKOS_ARCH_AMD_GFX1201 */
/* #undef KOKKOS_ARCH_AMD_GPU */
/* #undef KOKKOS_ARCH_VEGA */
/* #undef KOKKOS_ARCH_VEGA906 */
/* #undef KOKKOS_ARCH_VEGA908 */
/* #undef KOKKOS_ARCH_VEGA90A */
/* #undef KOKKOS_ARCH_NAVI */
/* #undef KOKKOS_ARCH_NAVI1030 */
/* #undef KOKKOS_ARCH_NAVI1100 */

/* #undef KOKKOS_IMPL_32BIT */
```

--- a/containers/src/Kokkos_DynamicView.hpp
+++ b/containers/src/Kokkos_DynamicView.hpp
@@ -515,7 +515,7 @@ class DynamicView : public Kokkos::ViewTraits<DataType, P...> {
         m_chunk_max((max_extent + m_chunk_mask) >>
                     m_chunk_shift)  // max num pointers-to-chunks in array
         ,
-        m_chunk_size(2 << (m_chunk_shift - 1)) {
+        m_chunk_size(1 << m_chunk_shift) {
     m_chunks = device_accessor(m_chunk_max, m_chunk_size);
 
     const std::string& label =

--- a/core/src/impl/Kokkos_TeamMDPolicy.hpp
+++ b/core/src/impl/Kokkos_TeamMDPolicy.hpp
@@ -103,7 +103,7 @@ KOKKOS_INLINE_FUNCTION auto nested_policy(
     TeamMDRangeMode<TeamMDRangeLastNestLevel::NotLastNestLevel,
                     TeamMDRangeParThread::ParThread,
                     TeamMDRangeParVector::NotParVector>,
-    TeamHandle const team, int count) {
+    TeamHandle const& team, int count) {
   return TeamThreadRange(team, count);
 }
 
@@ -112,7 +112,7 @@ KOKKOS_INLINE_FUNCTION auto nested_policy(
     TeamMDRangeMode<TeamMDRangeLastNestLevel::NotLastNestLevel,
                     TeamMDRangeParThread::NotParThread,
                     TeamMDRangeParVector::ParVector>,
-    TeamHandle const team, int count) {
+    TeamHandle const& team, int count) {
   return ThreadVectorRange(team, count);
 }
 
@@ -121,7 +121,7 @@ KOKKOS_INLINE_FUNCTION auto nested_policy(
     TeamMDRangeMode<TeamMDRangeLastNestLevel::NotLastNestLevel,
                     TeamMDRangeParThread::ParThread,
                     TeamMDRangeParVector::ParVector>,
-    TeamHandle const team, int count) {
+    TeamHandle const& team, int count) {
   return TeamVectorRange(team, count);
 }
 

